MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -863,27 +867,64 @@ void EnggDiffractionPresenter::inputChecksBeforeFitting(
                                 focusedRunNo);
   }
 
-  if (ExpectedPeaks.empty()) {
+  if (expectedPeaks.empty()) {
     g_log.warning() << "Expected peaks were not passed, via fitting interface, "
                        "the default list of "
                        "expected peaks will be utilised instead.\n";
   }
   bool contains_non_digits =
-      ExpectedPeaks.find_first_not_of("0123456789,. ") != std::string::npos;
+      expectedPeaks.find_first_not_of("0123456789,. ") != std::string::npos;
   if (contains_non_digits) {
-    throw std::invalid_argument("The expected peaks provided " + ExpectedPeaks +
+    throw std::invalid_argument("The expected peaks provided " + expectedPeaks +
                                 " is invalid, "
                                 "fitting process failed. Please try again!");
   }
 }
 
+std::string EnggDiffractionPresenter::validateFittingexpectedPeaks(
+    std::string &expectedPeaks) const {
+
+  if (!expectedPeaks.empty()) {
+
+    g_log.debug() << "Validating the expected peak list.\n";
+
+    auto *comma = ",";
+
+    for (size_t i = 0; i < expectedPeaks.size() - 1; i++) {
+      size_t j = i + 1;
+
+      if (expectedPeaks[i] == *comma && expectedPeaks[i] == expectedPeaks[j]) {
+        expectedPeaks.erase(j, 1);
+        i--;
+
```

Comment:
For later, this is the type of processing that can be done with `find_if`, `remove_if`, etc.